optimiser/optimisation_problem.py

```diff
@@ -0,0 +1,220 @@
+import pulp as pl
+from pulp import LpVariable
+
+
+def main():
+    # Define the problem
+    problem = pl.LpProblem("BatteryOptimisation", pl.LpMinimize)
+
+    # Time slices
+    T = range(1, 25)
+
+    TIMESLICE_SIZE = 0.5
+
+    # Renewable Generation Constants
+    MAX_RENEWABLE_GENERATION = 10  # kWh
+
+    # Battery Constants
+    MAX_BATTERY_DISCHARGE_RATE = 5
+    MAX_BATTERY_CHARGE_RATE = 5
+    MAX_BATTERY_CAP = 100
+
+    # Costs Constants
+    UNIT_PRICE_GRID_ELECTRICITY = 0.3
+    UNIT_PRICE_BATTERY_ELECTRICITY = 0
+    UNIT_PRICE_RENEWABLE_ELECTRICITY = 0
+    UNIT_PRICE_BATTERY_SALE_PRICE = 0.1
```

Comment:
Would be good to move constants into their own module - add this to a new issue ticket

Reply:
Added new issue https://github.com/davidwickh/Home-Battery-Optimiser/issues/4